lib/api/repository/login_repository.dart

```diff
@@ -0,0 +1,38 @@
+import 'package:flutter_survey/api/request/login_request.dart';
+import 'package:flutter_survey/api/service/login_service.dart';
+import 'package:flutter_survey/model/login_model.dart';
+import 'package:flutter_survey/api/exception/network_exceptions.dart';
+import 'package:flutter_survey/usecases/login_use_case.dart';
+import 'package:injectable/injectable.dart';
+import 'package:flutter_survey/env.dart';
+
+abstract class LoginRepository {
+  Future<LoginModel> doLogin({
+    required LoginInput input,
+  });
+}
+
+@Singleton(as: LoginRepository)
+class LoginRepositoryImpl extends LoginRepository {
+  final LoginService _loginService;
+
+  LoginRepositoryImpl(this._loginService);
+  @override
```

Comment:
we need 1 line break here
```suggestion
  LoginRepositoryImpl(this._loginService);

  @override
```